### DISCUSSION TITLE
Configure dynamic url for http sinks

### DISCUSSION THREAD

**Discussion by janiu-001** [vectordotdev/vector#16588] [Q&A] [answered]
For the http sinks, the url is like below, the url is static, and could we support dynamic url like: 
https://10.22.212.22:9000/:clientId/:deviceId, the clientId and deviceId is got from the messages, for example:https://10.22.212.22:9000/aaaa/id_1111111

![image](https://user-images.githubusercontent.com/19970581/221048114-f679acdf-b4c8-46b0-b4a9-69ca8b516324.png)

We are deciding on the final solution. Very appreciated for your quick response

**Comment by StephenWakely** [comment, marked as answer]
Currently this is not possible. We have an issue #1155 to track this. Feel free to add any context there.

**Comment by janiu-001** [comment]
@StephenWakely thanks very much for your response! we found another issues:

The vector will restart when it failed to connect to the source. How could solve this issue? By the way, we deploy with docker compose

And there are messages  lost when we restart the vector, the buffer type is disk
